File: tutopy_pkg/tutopy_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node


class MyNode(Node):
    def __init__(self):
        super().__init__("py_test_node")
        self._counter = 0
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        self._counter += 1
        self.get_logger().info("Hello " + str(self._counter))


def main(args=None):
    rclpy.init(args=args)

    node = MyNode()

    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: tutocpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("cpp_test_node"), counter(0)
    {
        timer = create_wall_timer(std::chrono::seconds(1), std::bind(&MyNode::timerCallback, this));
    }

private:
    rclcpp::TimerBase::SharedPtr timer;
    int counter;

    void timerCallback()
    {
        counter++;
        RCLCPP_INFO(get_logger(), "Hello %d", counter);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<MyNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}